packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -105,41 +104,51 @@ export function nodeDataToMapTree(
 ): MapTree {
 	assert(data !== undefined, 0x846 /* Cannot map undefined tree. */);
 
-	if (data === null) {
-		return valueToMapTree(data, allowedTypes);
-	}
-	switch (typeof data) {
-		case "number":
-		case "string":
-		case "boolean":
-			return valueToMapTree(data, allowedTypes);
-		default: {
-			if (isFluidHandle(data)) {
-				return valueToMapTree(data, allowedTypes);
-			} else if (Array.isArray(data)) {
-				return arrayToMapTree(data, allowedTypes);
-			} else if (data instanceof Map) {
-				return mapToMapTree(data, allowedTypes);
-			} else {
-				// Assume record-like object
-				return objectToMapTree(data as Record<string, InsertableContent>, allowedTypes);
-			}
-		}
+	const schema = getType(data, allowedTypes);
+
+	switch (schema.kind) {
+		case NodeKind.Leaf:
+			return leafToMapTree(data, schema, allowedTypes);
+		case NodeKind.Array:
+			return arrayToMapTree(data, schema);
+		case NodeKind.Map:
+			return mapToMapTree(data, schema);
+		case NodeKind.Object:
+			return objectToMapTree(data, schema);
+		default:
+			fail(`Unrecognized schema kind: ${schema.kind}.`);
 	}
 }
 
-function valueToMapTree(value: TreeValue, allowedTypes: ReadonlySet<TreeNodeSchema>): MapTree {
-	const mappedValue = mapValueWithFallbacks(value, allowedTypes);
+/**
+ * Transforms data under a Leaf schema.
+ * @param data - The tree data to be transformed. Must be a {@link TreeValue}.
+ * @param schema - The schema associated with the value.
+ * @param allowedTypes - The allowed types specified by the parent.
+ * Used to determine which fallback values may be appropriate.
+ */
+function leafToMapTree(
+	data: InsertableContent,
+	schema: TreeNodeSchema,
+	allowedTypes: ReadonlySet<TreeNodeSchema>,
+): MapTree {
+	assert(schema.kind === NodeKind.Leaf, "Expected a leaf schema.");
+	if (!isTreeValue(data)) {
+		// eslint-disable-next-line @typescript-eslint/no-base-to-string
+		throw new UsageError(`Input data is incompatible with leaf schema: ${data}`);
```

Comment:
Would be good to have a comment explaining why it makes sense to ignore this lint in this case. I think it makes sense, but could use a comment.